src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesLambdas.java

```diff
@@ -0,0 +1,19 @@
+package com.puppycrawl.tools.checkstyle.checks.coding.unnecessaryparentheses;
+
+public class InputUnnecessaryParenthesesLambdas {
+    int foo(int y) {
+        MathOperation addition = (x) -> x + x;
+        MathOperation addition2 = (x) -> { return x + x; };
+        MathOperation addition3 = x -> x + x;
+        MathOperation2 addition4 = (a, b) -> a + b;
```

Comment:
> Do you mean lambdas that are nested inside expressions?

inside Lambdas and probably few cases for inside expressions , just to make sure we do make difference between them.